File: src/WorkflowShell/mvn/MavenTest.rb
require_relative '../models/Command'

class MavenTest < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'mvntest'
    @command_description = 'Takes one argument, the name of a class which contains tests. Updates, then attempts to run all specified tests.'
    @command_usage = @command_string + " <Test Class Name>"
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments, false)

    test_name = parser_components.leftover_arguments.pop
    if test_name.nil?
      puts "You MUST specify the test name!"
      puts ""
      print_help(parser_components.basic_options_printer, nil)
    end

    command = "mvn -U -Dtest=#{test_name} test -Djava.awt.headless=true"
    run_shell_command(command, parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/mvn/MavenTest.js
const Command = require('../models/Command');

class MavenTest extends Command {

  static getString() {
    return 'mtest';
  }

  static getDescription() {
    return 'Takes one argument, the name of a class which contains tests. Updates, then attempts to run all specified tests.';
  }

  getUsage() {
    return `${MavenTest.getString()} <Test Class Name>`;
  }

  run(args) {
    const ok = super.run(args);
    if (!ok) {
      return false;
    }

    const testName = args._.shift();
    if (!testName) {
      console.log('You MUST specify the test name!');
      this.printHelp();
      return false;
    }

    const baseCommand = `mvn -U -Dtest=${testName} test -Djava.awt.headless=true`;
    return super.execute(baseCommand);
  }
}

module.exports = MavenTest;
